### DISCUSSION TITLE
Get the selected item key outside Madcad View ...

### DISCUSSION THREAD

**Discussion by jlvila** [jimy-byerley/pymadcad#25] [Q&A] [answered]
Hi,

I just integrated the View into PyQt5 app and then I need to get back the "key" of the selected shape into my app. I analyzed rendering.py and I don't see how to connect/intercept the inputEvent to get back the key (key = self.itemat(pos) ....)

thanks for your help
jl

**Comment by jimy-byerley** [comment, marked as answer]
Hello !

Glad you reached out to the code and understood its architecture :)
The way to receive events for an object in the scene (a `rendering.Display` instance) is to override its [`control`](https://pymadcad.readthedocs.io/en/latest/reference/rendering.html#madcad.rendering.Display.control) method
You can take a look at the way its used in
- [`kinematic.Solid.display`](https://github.com/jimy-byerley/pymadcad/blob/07adf336a67faac606b4575774a4b011d0cc658d/madcad/kinematic.py#L237)
- [`kinematic.Kinemanip`](https://github.com/jimy-byerley/pymadcad/blob/07adf336a67faac606b4575774a4b011d0cc658d/madcad/kinematic.py#L815)

`Display.control(view, key, sub, event)`  works just like a regular Qt event handler method, it receives an event and can choose to accept or ignore it. If it is ignored, the event will be sent over to the display's eventual children in the scene (corresponding to a sub-key), until it is accepted

You can alternatively override the view widget's method [`View.inputEvent`](https://pymadcad.readthedocs.io/en/latest/reference/rendering.html#madcad.rendering.View.inputEvent) to set a global behavior not specific to a `Display`. However I would recommend overriding the `View` only when overriding the `Display` is not sufficient.

Don't hesitate to tell me if you need more details
